core/eolearn/core/utilities.py

```diff
@@ -475,3 +476,113 @@ def constant_pad(X, multiple_of, up_down_rule='even', left_right_rule='even', pa
+
+
+    
+    :param data: input array
+    :type data: np.array
+    :param func: Mapping function that is applied on each slice. Outputs must have the same shape for every slice.
+    :type func: function np.array -> np.array
+    :param axis: Axis over which to map the function.
+    :type axis: int
+
+    :example:
+
+    >>> data = np.ones((5,10,10))
+    >>> func = lambda x: np.zeros((7,20))
+    >>> res = map_over_axis(data,func,axis=0)
+    >>> res.shape
+    (5, 7, 20)
+    """
+
+    # Move axis to front
+    data = np.moveaxis(data, axis, 0)
+    
+    res_mapped = [func(slice) for slice in data]
+    res = np.stack(res_mapped)
+    
+    # Move axis back if number of dimensions stays the same
+    if data.ndim == res.ndim:
+        res = np.moveaxis(res, 0, axis)
+    
+    return res
+
+def resize_images(data, new_size=None, scale_factors=None, anti_alias=True, interpolation='linear'):
+    """Resizes the image(s) acording to given size or scale factors.
+
+    To specify the new scale use one of `new_size` or `scale_factors` parameters.
+    
+    :param data: input image array
+    :type data: numpy array with shape (timestamps, height, width, channels), (height, width, channels), or (height, width)
+    :param new_size: New size of the data (width, height)
+    :type new_size: (int, int)
+    :param scale_factors: Factors (fx,fy) by which to resize the image
+    :type scale_factors: (float, float)
+    :param anti_alias: Use anti aliasing smoothing operation when downsampling.
+    :type anti_alias: bool
+    :param interpolation: Interpolation method used for resampling. One of 'nearest', 'linear', 'cubic'.
+    :type interpolation: bool
+    """
+
+    INTER_METHODS = {
+        'nearest': cv2.INTER_NEAREST, 
+        'linear': cv2.INTER_LINEAR,
+        'cubic': cv2.INTER_CUBIC
+    }
+
+    # Number of dimensions of input data
+    ndims = data.ndim
+
+    # Width and height axis for dimensionality
+    width_height_axis = {
+        2: (1,0),
+        3: (1,0),
+        4: (2,1)
+    }
+
+    # Old width and height
+    old_size = [data.shape[axis] for axis in width_height_axis[ndims]]
+
+    if new_size is not None and scale_factors is None:
+        scale_factors = [new/old for old, new in zip(old_size, new_size)]
+    elif scale_factors is not None and new_size is None:
+        new_size = [int(size * factor) for size,factor in zip(old_size, scale_factors)]
+    else:
+        raise ValueError('Exactly one of the arguments new_size, scale_factors must be given.')
+
+    if interpolation not in INTER_METHODS:
+        raise ValueError('Invalid interpolation method: %s' % interpolation)
+
+    interpolation_method = INTER_METHODS[interpolation]
+    downscaling = scale_factors[0] < 1 or scale_factors[1] < 1
+
+    def _resize2d(image):
+        # Perform anti-alias smoothing if downscaling
+        if downscaling and anti_alias:
+            sigmaX, sigmaY = [((1/s) - 1)/2 for s in scale_factors]
```

Comment:
Is it possible to give a reference to a source where the motivation and/or explanation is given why `sigma = ((1/s)-1)/2` is good option?

Reply:
This is the default value in the [skimage implementation of antialiasing](https://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.resize).

Here is an example using the old sigma values (sigma = 1/s). Band B02 is resized to the same resolution as B01 and compared with it. As you can see the blurring is excessive.

![orig_b01](https://user-images.githubusercontent.com/11095696/62520209-650e8400-b82d-11e9-8e5e-4efa9093c308.gif)

Here is an example using the skimage default sigma.

![skimage_b01](https://user-images.githubusercontent.com/11095696/62520440-e36b2600-b82d-11e9-8f4f-aaa41198b7fc.gif)

Reply:
Thanks for this info. Could you please add a comment to this code block describing from where this formula is taken?